openmc/mgxs/groups.py

```diff
@@ -77,6 +77,69 @@ def group_edges(self, edges):
         cv.check_greater_than('number of group edges', len(edges), 1)
         self._group_edges = np.array(edges)
 
+    @staticmethod
```

Comment:
I am curious about your thoughts as to why you went with this factory approach instead of adding these to the within the model subpackage.

Reply:
I don't have a strong reason -- I felt it was a natural pattern to use here, and it mirrors the factory pattern used for the `MGXS.get_mgxs(...)` static class method used by the `openmc.mgxs.Library` class to build `MGXS` instances for different reaction types and domains.